src/main/webapp/scripts/civic.js

```diff
@@ -2,19 +2,30 @@
  * Extract the government officials from the civic API response.
  */
 function extractOfficials(civicObj) {
-  let officials = new Object();
-  const levels = ["administrativeArea1", "administrativeArea2", "country", "international", "locality", "regional", "special", "subLocality1", "subLocality2"];
-  levels.forEach((level) => {
-    officials[level] = new Array();
-  });
+  let levelsArray = new Array();
+  const levels = {"international" : "International",
+                  "country" : "Country",
+                  "regional" : "Regional",
+                  "administrativeArea2" : "Administrative Area 2",
+                  "administrativeArea1" : "Administrative Area 1",
+                  "locality" : "Locality",
+                  "subLocality2" : "Sublocality 2",
+                  "subLocality1" : "Sublocality 1",
+                  "special" : "Special",};
+  for (const level in levels) {
+    let levelsObj = new Object();
+    levelsObj.levelType = level;
+    levelsObj.levelName = levels[level]
+    levelsObj.officials = new Array();
+    levelsArray.push(levelsObj);
+  }
   civicObj.offices.forEach((officeObj) => {
```

Comment:
can these two foreach's be accomplished with a regular for loop like you have above, or is there a specific reason why you're doing it this way here¿

Reply:
I prefer using forEach because I believe is more readable, but I had to use a regular for loop above because I was iterating over an object and not an array.